src/main/java/de/tum/in/www1/artemis/service/user/UserCreationService.java

```diff
@@ -165,7 +165,8 @@ public User createUser(ManagedUserVM userDTO) {
                     .collect(Collectors.toSet());
             user.setAuthorities(authorities);
         }
-        String encryptedPassword = passwordService.encodePassword(userDTO.getPassword() == null ? RandomUtil.generatePassword() : userDTO.getPassword());
+        String password = userDTO.getPassword() == null ? RandomUtil.generatePassword() : userDTO.getPassword();
+        String encryptedPassword = passwordService.hashPassword(password);
         user.setPassword(encryptedPassword);
```

Comment:
```suggestion
        String passwordHash = passwordService.hashPassword(password);
        user.setPassword(passwordHash);
```
More occurrences in other files as well.